src/main/java/pro/belbix/ethparser/web3/prices/parser/PriceLogParser.java

```diff
@@ -155,7 +155,10 @@ private boolean skipSimilar(PriceDTO dto) {
 
   private boolean isValidSource(PriceDTO dto) {
     String currentLpName = ContractUtils
-        .findUniPairNameForTokenName(dto.getToken(), dto.getBlock());
+        .findUniPairNameForTokenName(dto.getToken(), dto.getBlock()).orElse(null);
```

Comment:
not so important but you changed old behavior
better keep old logic when refactoring something (throw a concrete exception in this case)